pkg/ioutil/pagewriter.go

```diff
@@ -38,9 +36,10 @@ type PageWriter struct {
 	bufWatermarkBytes int
 }
 
-func NewPageWriter(w io.Writer, pageBytes int) *PageWriter {
+func NewPageWriter(w io.Writer, pageOffset, pageBytes int) *PageWriter {
```

Comment:
`NewPageWriter(w io.Writer, pageBytes, pageOffset int) *PageWriter` -- order the arguments by importance; the page size is more important to the operation of the writer than the offset since the offset can be dropped in the default cases (i.e., a new file) and the writer will work, but the page size must always be provided.